src/llscan.cc

```diff
@@ -0,0 +1,470 @@
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include <algorithm>
+#include <fstream>
+#include <vector>
+
+#include <lldb/API/SBExpressionOptions.h>
+
+#include "src/llnode.h"
+#include "src/llv8.h"
+#include "src/llv8-inl.h"
+#include "src/llscan.h"
+
+namespace llnode {
+
+// Defined in llnode.cc
+extern v8::LLV8 llv8;
+
+struct MemoryRange {
+    uint64_t start;
+    uint64_t length;
+    MemoryRange* next;
+};
+
+static MemoryRange* ranges = nullptr;
```

Comment:
This whole thing may be instantiated globally in `llnode.cc`, just as `LLV8`. The same caching by comparing that `target_` to current should probably apply.